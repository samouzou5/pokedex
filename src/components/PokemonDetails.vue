<template>
  <div class="pokemon-details">
    <template v-if="pokemon">
      <div class="id">
        <span class="symbol">#</span>
        <span class="number">{{ pokemon.id }}</span>
      </div>
      <div class="name">{{ pokemon.name }}</div>
      <BaseImage
        :src="pokemon.image"
        :alt="`Pokemon ${pokemon.id}`"
        class="preview"
      />
      <button class="toggle-favorite" @click="toggleFavorite()">Toggle Favorite</button>
    </template>
    <div v-else class="empty">
      Pokemon not found
    </div>
  </div>
</template>

<script>
import pokemons from '@/assets/pokemons.json'
export default {
  props: {
    id: {
      type: [Number, String],
      required: false
    }
  },
  computed: {
    pokemon () {
      const id = parseInt(this.id)
      return pokemons.find(
        pokemon => pokemon.id === id
      )
    }
  },

  created () {
    console.log(this.$route.params)
  },

  methods: {
    toggleFavorite () {
      this.$emit('toggle-favorite', this.id)
    }
  }
}
</script>
<style lang="stylus" scoped>
@import '~@/style/imports'
.pokemon-details
  background $color-grey
  border-radius $br
  margin $padding
  padding $padding
.id
  .symbol
    color $color-grey-dark
.name
  font-size 2em
  font-weight lighter
.preview
  width 192px
  height @width
  >>> .image
    image-rendering pixelated
</style>
