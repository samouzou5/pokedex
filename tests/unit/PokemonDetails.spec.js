import PokemonDetails from '@/components/PokemonDetails'
import { mount } from '@vue/test-utils'

describe('PokemonDetails', () => {
  const mocks = {
    $route: { params: { id: 3 } }
  }

  const stubs = {
    'BaseImage': '<div data-stub-base-image v-bind="$attrs"></div>'
  }
  test('snapshot', () => {
    const wrapper = mount(PokemonDetails, {
      propsData: {
        id: 3
      },
      mocks,
      stubs
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.name')).toBe('Venusaur')
  })

  test('snapshot: pokemon not found', () => {
    const wrapper = mount(PokemonDetails, {
      propsData: {
        id: 'foo'
      },
      mocks,
      stubs
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.empty').text()).toBe('Pokemon not found')
  })

  test('event: toggle-favorite', () => {
    const wrapper = mount(PokemonDetails, {
      propsData: {
        id: 2
      },
      mocks,
      stubs
    })
    wrapper.find('.toggle-favorite').trigger('click')
    expect(wrapper.emitted('toggle-favorite')).toBeTruthy()
    expect(wrapper.emitted('toggle-favorite'))[0].toEqual([2])
  })
})
