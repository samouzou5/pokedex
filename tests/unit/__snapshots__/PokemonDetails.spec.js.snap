// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetails snapshot 1`] = `
<div class="pokemon-details">
  <div class="id"><span class="symbol">#</span> <span class="number">3</span></div>
  <div class="name">Venusaur</div>
  <div data-stub-base-image="" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png" alt="Pokemon 3" class="preview"></div>
  <button class="toggle-favorite">Toggle Favorite</button>
</div>
`;

exports[`PokemonDetails snapshot: pokemon not found 1`] = `
<div class="pokemon-details">
  <div class="empty">
    Pokemon not found
  </div>
</div>
`;
